src/common/commandExecutor.ts

```diff
@@ -152,11 +126,17 @@ export class CommandExecutor {
         });
 
         result.outcome = result.outcome.then(
-            () =>
-                Log.logCommandStatus(commandWithArgs, CommandStatus.End),
+            () => {
+                if (waitForExit) {
+                    Log.logCommandStatus(commandWithArgs, CommandStatus.End);
+                } else {
+                    Q.delay(300).done(() => {
+                        return result;
```

Comment:
Returning a value in a done() handler won't have any effect I think; what is this supposed to do?